drivers/firmware/mediatek/mtk-adsp-ipc.c

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2021 MediaTek Corporation. All rights reserved.
+ * Author: Allen-KH Cheng <allen-kh.cheng@mediatek.com>
+ */
+
+#include <linux/firmware/mediatek/mtk-adsp-ipc.h>
+#include <linux/kernel.h>
+#include <linux/mailbox_client.h>
+#include <linux/module.h>
+#include <linux/of_platform.h>
+#include <linux/platform_device.h>
+#include <linux/slab.h>
+
+int mtk_adsp_ipc_send(struct mtk_adsp_ipc *ipc, unsigned int idx, uint32_t msg)
+{
+	struct mtk_adsp_chan *dsp_chan;
+	int ret;
+
+	if (idx >= MTK_ADSP_MBOX_NUM)
+		return -EINVAL;
+
+	dsp_chan = &ipc->chans[idx];
+	ret = mbox_send_message(dsp_chan->ch, &msg);
```

Comment:
if you initialize `ret = 0` you can just return here and remove the if statement

Reply:
Hi Curtis, 

Sorry I am not sure what you means. 

If  I just return here using mbox_send_message, it seems no need res variable.

Reply:
ah so it returns > 0 on success @plbossart FYI as i saw you had the same comment on the message board

Reply:
ok, then you definitively need a comment to make sure the handling of positive values is captured